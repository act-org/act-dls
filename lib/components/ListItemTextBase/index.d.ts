/**
 * @prettier
 */
import * as React from 'react';
import { ListItemTextProps } from '@material-ui/core';
export declare type Props = ListItemTextProps;
declare const ListItemTextBase: React.FC<Props>;
export default ListItemTextBase;
//# sourceMappingURL=index.d.ts.map