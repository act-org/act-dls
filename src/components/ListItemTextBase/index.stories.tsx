/**
 * @prettier
 */

import * as React from 'react';
import { boolean, text } from '@storybook/addon-knobs';

import ListItemTextBase from '.';

export const Base = (): React.ReactElement<any> => (
  <ListItemTextBase
    disableTypography={boolean('Disable Typography?', false)}
    inset={boolean('Inset?', false)}
    primary={text('Primary Text', 'List item primary text')}
    secondary={text('Secondary Text', 'List item secondary text')}
  >
    ListItemText
  </ListItemTextBase>
);

export default {
  component: ListItemTextBase,
  parameters: {
    info: {
      text: 'This is the base variant of the ListItemText component.',
    },
  },
  title: 'Atoms/ListItemText',
};
