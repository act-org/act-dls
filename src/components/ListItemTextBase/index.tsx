/**
 * @prettier
 */

import * as React from 'react';
import { ListItemText, ListItemTextProps } from '@material-ui/core';

export type Props = ListItemTextProps;

const ListItemTextBase: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => <ListItemText {...props} />;

export default ListItemTextBase;
