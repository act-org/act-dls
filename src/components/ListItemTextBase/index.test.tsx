/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import ListItemTextBase from '.';

describe('ListItemTextBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <ListItemTextBase primary="Primary text" secondary="Secondary text" />,
    );

    expect(container).toMatchSnapshot();
  });
});
