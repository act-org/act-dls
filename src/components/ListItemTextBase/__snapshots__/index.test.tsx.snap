// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemTextBase matches the snapshot 1`] = `
<div>
  <div
    class="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      Primary text
    </span>
    <p
      class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
    >
      Secondary text
    </p>
  </div>
</div>
`;
